R/auditCommonFunctions.R

```diff
@@ -2304,6 +2304,29 @@
                                                                 "sampleIndicatorColumn"))
     jaspResults[["sampleIndicatorColumn"]]$setNominal(sampleIndicatorColumn)
   }
+  
+  # Export sample
+  if(options[["exportSample"]] && options[["file"]] != ""){
+    sampleExport <- data.frame(row = parentState[["rowNumber"]])
+    sampleExport <- cbind(sampleExport, count = parentState[["count"]])
+    sampleExport <- cbind(sampleExport, id = parentState[[.v(options[["recordNumberVariable"]])]])
+    colnames(sampleExport) <- c("Row number", decodeColNames(options[["sampleIndicatorColumn"]]), decodeColNames(options[["recordNumberVariable"]]))
+    if(options[["monetaryVariable"]] != ""){
+      sampleExport <- cbind(sampleExport, ist = parentState[[.v(options[["monetaryVariable"]])]])
+      colnames(sampleExport)[length(colnames(sampleExport))] <- decodeColNames(options[["monetaryVariable"]])
+    }
+    if(options[["rankingVariable"]] != ""){
+      sampleExport <- cbind(sampleExport, parentState[[.v(options[["rankingVariable"]])]])
+      colnames(sampleExport)[length(colnames(sampleExport))] <- decodeColNames(options[["rankingVariable"]])
+    }
+    if(length(unlist(options[["additionalVariables"]])) >= 1 && unlist(options[["additionalVariables"]]) != ""){
+      sampleExport <- cbind(sampleExport, parentState[[.v(unlist(options[["additionalVariables"]]))]])
+      colnames(sampleExport)[(length(colnames(sampleExport)) - length(unlist(options[["additionalVariables"]]))):length(colnames(sampleExport))] <- decodeColNames(unlist(options[["additionalVariables"]]))
+    }
+    sampleExport <- cbind(sampleExport, rep(NA, nrow(sampleExport)))
+    colnames(sampleExport)[length(colnames(sampleExport))] <- "Soll"
+    utils::write.csv2(x = sampleExport, file = options[["file"]], row.names = FALSE, na = "", dec = ",", sep = ";", quote = FALSE)
```

Comment:
Wouldn't it be nice to have this be decided by the locale? Perhaps add an option for this to be changed to standard csv files.

Reply:
I was thinking that as well, but implemented it like this since it's the dutch standard and uvb will be testing this.

Reply:
Or just change it to regular csv?